### PR TITLE
Add rebar3 manager

### PR DESCRIPTION
**Pull request by ericmj**
Compilation doesn't work yet, we are waiting for rebar3 to add a way to specify dependency build directory. We may use an archive instead of an escript for rebar3 if that makes it easier. /cc @tsloughter

### CODE REVIEW

File: lib/mix/lib/mix/rebar.ex [hunk 1]
```diff
@@ -162,4 +174,38 @@ defmodule Mix.Rebar do
       rebar
     end
   end
+
+  def apply_overrides(app, config, overrides) do
```
Comment (josevalim) [inline lib/mix/lib/mix/rebar.ex]: `@doc` missing.

**Comment by josevalim** [comment]
:heart:

**Comment by josevalim** [comment]
I am adding it to the v1.2 milestone because folks are asking for this more and more.

**Comment by ericmj** [comment]
This is currently blocked on https://github.com/rebar/rebar3/issues/863.

**Comment by ericmj** [comment]
This is ready for review.

If we merge this for 1.2.0 we should treat the rebar3 support as experimental because it will likely have hard to find bugs due to the nature of build tools calling each other. Additionally rebar3 is not stable yet so it can have breaking changes.

**Pull request merged**